File: ready-set-boole/src/bin/exercices/ex01.rs
use super::ex00::adder;

pub fn multiplier(a: u32, b: u32) -> u32 {
    let mut a = a;
    let mut b = b;
    let mut result: u32 = 0;

    if a == 0 {
        return 0;
    }
    while b != 0 {
        if b & 1 != 0 {
            result = adder(a, result);
        }
        a = a << 1;
        b = b >> 1;
    }
    return result;
}

File: fr/rvander/ready_set_boole/Ex01.java
package fr.rvander.ready_set_boole;


public class Ex01 {

	public static int multiplier(int a, int b) {
		int result = 0;

		if (a == 0) {
			return 0;
		}
		while (b != 0) {
			if ((b & 1) != 0) {
				result = Ex00.adder(a, result);
			}
			a = a << 1;
			b = b >>> 1;
		}
		return result;
	}
}